src/api/status/src/js/github-stats.js

```diff
@@ -1,55 +1,50 @@
-const { fetch } = require('@senecacdot/satellite');
-
+const { logger } = require('@senecacdot/satellite');
+const octokit = require('./octokit');
 const redis = require('../redis');
 
 const cacheTime = 60 * 10; // 10 mins of cache time
 
-const fetchGitHubApi = (owner, repo, path) =>
-  fetch(`https://api.github.com/repos/${owner}/${repo}/${path}`, {
-    headers: { Accept: 'application/vnd.github.v3+json' },
-  });
-
 const getStatsParticipation = async (owner, repo) => {
   // get weekly commits for last year: https://docs.github.com/en/rest/reference/repos#get-the-weekly-commit-count
-  const res = await fetchGitHubApi(owner, repo, 'stats/participation');
-  const participationResponse = await res.json();
-  if (!res.ok) {
-    throw new Error(`[Code ${res.status}] - ${participationResponse.message}`);
-  }
+  const participationResponse = await octokit.request(
+    'GET /repos/{owner}/{repo}/stats/participation',
+    {
+      owner,
+      repo,
+    }
+  );
 
   return {
     weeklyCommits: {
-      commits: participationResponse.all[participationResponse.all.length - 1],
+      commits: participationResponse.data.all[participationResponse.data.all.length - 1],
     },
     yearlyCommits: {
-      commits: participationResponse.all.reduce((a, b) => a + b),
+      commits: participationResponse.data.all.reduce((a, b) => a + b),
     },
   };
 };
 
 const getStatsCodeFrequency = async (owner, repo) => {
   // get weekly commits activity: https://docs.github.com/en/rest/reference/repos#get-the-weekly-commit-activity
-  const res = await fetchGitHubApi(owner, repo, 'stats/code_frequency');
-  const codeFrequencyResponse = await res.json();
-
-  if (!res.ok) {
-    throw new Error(`[Code ${res.status}] - ${codeFrequencyResponse.message}`);
-  }
-
-  const [, linesAdded, linesRemoved] = codeFrequencyResponse[codeFrequencyResponse.length - 1];
+  const codeFrequencyResponse = await octokit.request(
+    'GET /repos/{owner}/{repo}/stats/code_frequency',
+    {
+      owner,
+      repo,
+    }
+  );
+  const [, linesAdded, linesRemoved] =
```

Comment:
Why do we need this extra comma ?

Reply:
It's a way to skip an earlier value:

```js
const [, b, c] = [1, 2, 3];
// b = 2
// c = 3
```

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment